### DISCUSSION TITLE
Is it possible in architecture tests to check that a dependency does not exist?

### DISCUSSION THREAD

**Discussion by ravenfeld** [LemonAppDev/konsist#810] [Ask a Question]
### Wanted end result.

Hi, I'd like to write something like :

```
domain.dependsOnNothing()
ui.dependsOn(domain)
data.dependsOn(domain)

ui.isnotdependsOn(data)
data.isnotdependsOn(ui)
``` 

I'd just like to check that there are no dependencies in some of them.

Thanks in advance

### What you tried so far.

So far I've only found 

`dependsOnNothing` or `dependsOn`

**Comment by nateridderman** [comment]
As a partial solution, it would be nice to be able to write:
`assertThrows<KoAssertionFailedException> { layer1.dependsOn(layer2) }`

However, the assertions don't actually happen in `dependsOn`. They seem to happen later in KoArchitectureAssertionCore.

**Reply by igorwojda** [reply]
Not yet, but we have ticket to cover this case https://lemonappdev.atlassian.net/browse/KON-623

**Comment by nataliapeterwas** [comment]
Will be added in the next release: [PR](https://github.com/LemonAppDev/konsist/pull/1233) :)

**Comment by nataliapeterwas** [comment]
Konsist `0.16.0` has been released! This release includes this improvement :)
